### PR TITLE
Add CI job running test suite with `async-proofs on`

### PR DESCRIPTION
**Pull request by maximedenes**
This is based on #9072 and #9087.

Will probably require fixing some STM problems revealed here.

### CODE REVIEW

File: test-suite/Makefile [hunk 1]
```diff
@@ -36,15 +36,16 @@ include ../Makefile.common
 # easily overridden
 LIB := ..
 BIN := $(shell cd ..; pwd)/bin/
+COQFLAGS := ""
```
Comment (SkySkimmer) [inline test-suite/Makefile]: Pretty sure this is broken. Use `COQFLAGS?=` instead.

**Comment by maximedenes** [comment]
The current build trace seems to reveal a race, maybe @gares or @ejgallego know about it. Is it the one you wrote a comment about in the code, Emilio?

```
31358:master:4] Uncaught exception in worker manager: 
                Anomaly
                "The global environment cannot be accessed during parsing."
                Please report at http://coq.inria.fr/bugs/.
```

**Comment by ejgallego** [comment]
> Is it the one you wrote a comment about in the code, Emilio?

We had a long discussion about this check at Nice, this check was introduced by @gares .

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -132,7 +132,7 @@ after_script:
     - BIN=$(readlink -f ../_install_ci/bin)/
     - LIB=$(readlink -f ../_install_ci/lib/coq)/
     - export OCAMLPATH=$(readlink -f ../_install_ci/lib/):"$OCAMLPATH"
-    - make -j "$NJOBS" BIN="$BIN" LIB="$LIB" all
+    - make -j "$NJOBS" BIN="$BIN" LIB="$LIB" COQFLAGS="${COQFLAGS}" all
```
Comment (ejgallego) [inline .gitlab-ci.yml]: Drop the `{}` from the quoting.

**Review by ejgallego — APPROVED** [review]
Seems OK, but needs to work.

**Comment by maximedenes** [comment]
I need help (probably from @gares and @SkySkimmer) to fix the universe issues. It is probably revealing some serious problems.

**Comment by maximedenes** [comment]
That is, we need to fix #9093.

**Comment by maximedenes** [comment]
Hum the last run shows more problems, something must be off.

**Comment by maximedenes** [comment]
@SkySkimmer when I download the artifact, I don't see the `bugs/opened` files (I do see the other ones, though). Is this an infrastructure bug?

**Comment by SkySkimmer** [comment]
Yeah we explicitly exclude the opened/ logs. I think I had a misunderstanding about how they work when I wrote the script or some such issue. If you just remove `'(' -path ./bugs/opened -prune ')' -o` from report.sh it should work (maybe? I still don't fully understand the opened/ tests)

**Comment by maximedenes** [comment]
@gares the regression in the test suite was introduced by 4838146ee993a3bafb6a359dff529aae815cfff6

Some files contain non-terminated proofs that trigger anomalies. But since that commit, the anomaly has been replaced by an error saying that the proof is not closed.

What seems strange is that `coqc -async-proofs on` now has a behavior which is different from `coqc` but also different from opening `coqide` and jumping to the end of the file. Is that expected?

**Comment by maximedenes** [comment]
I made the tests more robust, it should go through now.

Dear coqbot, can you trigger a CI build even if you are not in a good mood today? I'll send you chocolates for Christmas.

**Comment by maximedenes** [comment]
The last remaining problem is #3934. Working on a fix.

**Comment by maximedenes** [comment]
Finally, everything passed! @ejgallego can we merge this, to make sure we don't introduce STM-related regressions? Thanks a lot!

**Comment by maximedenes** [comment]
Thanks!

**Pull request merged**